src/Microsoft.Azure.SignalR/HubHost/ServiceConnectionManager.cs

```diff
@@ -29,5 +30,13 @@ public void AddServiceConnection(ServiceConnection serviceConnection)
             var index = StaticRandom.Next(_serviceConnections.Count);
             await _serviceConnections[index].WriteAsync(serviceMessage);
         }
+
+        public async Task WriteAsync(ServiceMessage serviceMessage, int index)
```

Comment:
>WriteAsync [](start = 26, length = 10)

how about also hiding the hashing details and expose a `WriteWithGroup`, so that the manager can decide what algorithm to use, e,g, `name.GetHashCode % 5`, or maybe other partitioning algorithms?

Reply:
Make sense. Will be consistent with the original `WriteAsync`.
I plan to update it to `public async Task WriteAsync(string partitionKey, ServiceMessage serviceMessage)`.

Reply:
I am not sure. So the caller decides what is the `partitionKey`, is it too much for the caller?

Reply:
Essentially the same as `WriteWithGroup`, but more general. `partitionKey` just means that the same key will go through the same service connection. Caller doesn't have to care the details of mapping partition keys.
In this case, when the same group name is passed in as partition key, the service message will be sent out via the same service connection.

Reply:
I thin partition key is fine.